common/lib/azure/storage/common/service/storage_service.rb

```diff
@@ -239,8 +239,8 @@ class << self
 
         # Registers the callback when sending the request
         # The headers in the request can be viewed or changed in the code block
-        def register_request_callback
-          @request_callback = Proc.new
+        def register_request_callback(&block)
+          @request_callback = proc &block
```

Comment:
Fixes Ruby's `>= 3.0`: `ArgumentError (tried to create Proc object without a block)`